buildtest/system.py

```diff
@@ -23,20 +24,55 @@ def __init__(self):
         """Constructor method for BuildTestSystem(). Defines all system configuration using
            class variable **system** which is a dictionary
         """
+        self.logger = logging.getLogger(__name__)
+        self.logger.debug("Starting System Compatibility Check")
         self.init_system()
-        self.system["SYSTEM"] = platform.system()
+        self.system["platform"] = platform.system()
         self.scheduler = self.check_scheduler()
         self.check_lmod()
 
+        if self.system["platform"] != "Linux":
+            print("System must be Linux")
+            sys.exit(1)
+
+        self.logger.debug("Finished System Compatibility Check")
+
     def init_system(self):
         """Based on the module "distro" set the linux distrubution name and version
         """
-        self.system["OS_NAME"] = distro.linux_distribution(
+
+        self.system["os"] = {}
+
+        self.system["os"]["name"] = distro.linux_distribution(
             full_distribution_name=False
         )[0]
-        self.system["OS_VERSION"] = distro.linux_distribution(
+        self.system["os"]["version"] = distro.linux_distribution(
             full_distribution_name=False
         )[1]
+        self.system["os"]["major_version"] = distro.major_version(best=True)
+        self.system["os"]["minor_version"] = distro.major_version(best=True)
+        self.system["os"]["build_number"] = distro.build_number(best=True)
+        self.system["env"] = dict(os.environ)
+        self.system["python"] = shutil.which("python") or None
+
+        self.logger.info(f"Operating System: {self.system['os']['name']}")
+        self.logger.info(f"Operating System Version: {self.system['os']['version']}")
+        self.logger.info(
+            f"Operating System Major Version: {self.system['os']['major_version']}"
```

Comment:
Is there any reason to not assemble a complete version string and have a single, clean message for OS version over printing a line for each?

Reply:
no particular reason, just providing extra detail. I mean this is sufficient also at the moment it not providing any extra value other that additional detail. 

```
 distro.linux_distribution()
('Red Hat Enterprise Linux Server', '7.6', 'Maipo')
```

Reply:
Yes let’s do that, it’s more human readable as a single line. If you want to join with a space to avoid the extra layer of data structure that works too.

Reply:
okay this is fixed now it looks like this now

```
(buildtest-framework) ssi29@ag-mxg-hulk090> cat buildtest.log | grep Operating
2020-04-08 22:00:23,242 [system.py:49 - init_system() ] - [DEBUG] Operating System: Red Hat Enterprise Linux Server 7.6 Maipo

```

Reply:
Much improved.